src/pages/auth/ForgotPassword.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { Trans, t } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+
+import Button from 'components/button/Button';
+import TextField from 'components/textfield/TextField';
+import { Panel, PanelContent } from 'components/panel/Panel';
+import { SignUp, TitleMessage } from './Utilities';
+import { ReactComponent as HandImage } from '../home/assets/images/forgot-password.svg';
+
+class ForgotPassword extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getAccountInfo = this.getAccountInfo.bind(this);
```

Comment:
Don't we still have to bind to the constructor? Or can you really use arrow functions for this?

Reply:
Arrow function will do this binding for us.

Reply:
@tureey - Interesting point of view since React official documentation doesn't talk about this approach. I did a bit of reading about this and tried a few things locally. Here is my comparison between the two approaches

Arrow functions:
Pros:
1) No need to bind the function to the constructor

Cons:
When you use arrow functions, it does not get defined on the ProtoType. This has some consequences:
1) Child classes cannot call by, super.foo().
2) Overriding foo(), results in the creation of the function twice(one for super, one for derived)
3) This results in relatively slower performance compared to prototype approach.
4) It is hard to spy/stub/mock for testing while prototype functions can be easily mocked before instantiation.

I think the cons outweigh the pros here. I would stick with the current approach. But, I am interested in hearing more from you.

Reply:
About your points:
1 and 2. I never had to use them using React.
3. I don't know how much it really affects. Could you share some links talking about it, please? Because maybe I will it stop doing it.
4. I have never been a big fan of testing UI

Reply:
Well, even knowing this I would go for arrow functions, but just a total personal opinion.

Since you are using a callback to use `saveEmail` as `event => this.saveEmail(event)`. You're doing exactly the same than an arrow function, re assigning the function execution in run time.

So, you're having the same performance "issue". To evade this, you can `bind()` the function to the prototype as you commented but then to execute the function, you should just pass the reference to this as: ` onChange={this.saveEmail}`

Reply:
Good catch. I modified it as per your suggestion. Thanks for pointing out. At this point, I am convinced about the arrow functions approach especially because it is cleaner. But, I would like others to pitch in and share their opinion too. If all of us agree with that approach, maybe we can open up a story to convert all components to use the arrow functions approach to have it uniform. :)

Reply:
Sure!! I would say though that it's a standard in the React community